app/components/floating_banner/floating_banner.tsx

```diff
@@ -0,0 +1,242 @@
+    const testID = isTop ? 'floating-banner-top-container' : 'floating-banner-bottom-container';
+    const baseBottomOffset = Platform.OS === 'android' ? FLOATING_BANNER_BOTTOM_OFFSET_PHONE_ANDROID : FLOATING_BANNER_BOTTOM_OFFSET_PHONE_IOS;
+    const bottomOffset = isTablet ? (baseBottomOffset + FLOATING_BANNER_TABLET_EXTRA_BOTTOM_OFFSET) : baseBottomOffset;
+    const containerStyle = isTop ? [styles.containerBase, styles.topContainer] : [
+        styles.containerBase,
+        styles.bottomContainer,
+    ];
+
+    const animatedStyle = useAnimatedStyle(() => {
+        if (isTop) {
+            return {};
+        }
+
+        if (Platform.OS === 'android') {
+            return {bottom: withTiming(FLOATING_BANNER_BOTTOM_OFFSET_WITH_KEYBOARD_ANDROID, {duration: 250})};
+        }
+
+        return {bottom: withTiming((keyboardHeight > 0 ? FLOATING_BANNER_BOTTOM_OFFSET_WITH_KEYBOARD_IOS : bottomOffset) + keyboardHeight, {duration: 250})};
+    }, [keyboardHeight, isTop, bottomOffset]);
+
+    return (
+        <Animated.View
+            testID={testID}
+            style={[containerStyle, animatedStyle]}
+        >
+            <GestureHandlerRootView style={styles.gestureHandler}>
+                {sectionBanners.map((banner, index) => (
+                    <AnimatedBannerItem
+                        key={banner.id}
+                        banner={banner}
+                        index={index}
+                        isTop={isTop}
+                        isTablet={isTablet}
+                        insets={insets}
+                        executeBannerAction={executeBannerAction}
+                        dismissBanner={dismissBanner}
+                    />
+                ))}
+            </GestureHandlerRootView>
+        </Animated.View>
+    );
+};
+
+const FloatingBanner: React.FC<FloatingBannerProps> = ({banners, onDismiss}) => {
+    const executeBannerAction = (banner: BannerConfig) => {
+        if (banner.onPress) {
+            banner.onPress();
+        }
+    };
+
+    const dismissBanner = (banner: BannerConfig) => {
+        onDismiss(banner.id);
+        if (banner.onDismiss) {
+            banner.onDismiss();
+        }
+    };
+
+    if (!banners || !banners.length) {
+        return null;
+    }
+
+    const topBanners = banners.filter((banner) => (banner.position || 'top') === 'top');
+    const bottomBanners = banners.filter((banner) => banner.position === 'bottom');
+
+    return (
+        <>
+            <BannerSection
+                sectionBanners={topBanners}
+                position='top'
+                executeBannerAction={executeBannerAction}
+                dismissBanner={dismissBanner}
+            />
+            <BannerSection
+                sectionBanners={bottomBanners}
+                position='bottom'
+                executeBannerAction={executeBannerAction}
+                dismissBanner={dismissBanner}
+            />
+        </>
+    );
+};
+
+const styles = StyleSheet.create({
```

Comment:
styles should be defined before the components as we have in any other component

Reply:
This one has been refactored out significantly, but I've moved all the stylesheet to above the component itself.